### PR TITLE
Allow the Emulator user to specify a User ID 

### LINKED ISSUE #1344: Allow the Emulator user to specify a User ID to be used for conversations
see https://github.com/Microsoft/BotFramework-Emulator/issues/197 & https://github.com/Microsoft/BotFramework-Emulator/issues/1254

### PR DESCRIPTION
**Pull request by Aliandi**
Fixes #1344
### Description
This PR allows the Emulator users to specify a User ID which will be used within conversations.

#### Functionality changes
To implement this feature we made the following changes:

- Add an input to the Emulator Settings so that the user can configure a GUID
- Change the startNewConversation method to choose between a randomly generated UserId, the UserId set in the props or the UserId set by the user.'
- When the Use custom id checkbox is set to false, the input is disabled and cleaned. When true, it is a required field and it gets enabled.
- The User ID field has a validation so that the user is not able to enter anything other than a GUID.
- Restarting a conversation with the same userId will choose between the one set by the user or the one set by default (if the first one is empty).
- Restarting a conversation with a new userId will generate a random ID the same way it was doing it until now.
- Remove _Restart Conversation_ Split Button
- Add two new buttons _Restart with Same User Id_ and _Restart with New User Id_ to replace _Restart Conversation_ Split Button.

#### UI Changes

**Before**
imagen
imagen

**After**
imagen
imagen

### Testing
**Manual testing**
BETranscript-Test

**Automated testing**
Added automated tests for the new settings options.

imagen

### CODE REVIEW

File: packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx [hunk 1]
```diff
@@ -92,8 +92,18 @@ export class AppSettingsEditor extends React.Component<AppSettingsEditorProps, A
     this.props.getFrameworkSettings();
   }
 
+  private usingCustomId() {
+    return this.state.userGUID ? true : this.state.useCustomId;
+  }
+  public GUIDValidator(guid: string) {
```
Comment (justinwilaby) [inline packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx]: there are currently no requirements for the user id to be a valid GUID. Some channels will use an email address for the user id others will use a big int. This constraint should be removed to allow for these differences.

Comment (justinwilaby) [inline packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx]: there are currently no requirements for the user id to be a valid GUID. Some channels will use an email address for the user id others will use a big int. This constraint should be removed to allow for these differences.

Comment (Aliandi) [reply packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx]: Ok! We understood from the issue that it had to be a valid GUID. We are changing it now!

**Comment by denscollo** [comment]
> 
> 
> Great job! 🎉

Thanks! 🎉

**Comment by cwhitten** [comment]
Hi @Aliandi and @denscollo:

Justin's in-flight work in https://github.com/Microsoft/BotFramework-Emulator/pull/1400 had problems reconciling with some of the changes made in this diff, so we reverted it in the hopes that when his branch lands @Aliandi can reissue the pull request against master again.

**Comment by cwhitten** [comment]
@denscollo @Aliandi this is ready to be re-opened against master. There has been a refactor to some of the code-paths involved so some changes will be necessary.

**Pull request merged**